src/mpid/ch4/src/ch4_comm.h

```diff
@@ -209,6 +210,22 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_Comm_create_hook(MPIR_Comm * comm)
             default:
                 MPIDIU_avt_add_ref(MPIDI_COMM(comm, local_map).avtid);
         }
+
+        /* translate info hints here */
+        resource = properties = 0;
+        if (MPIDI_CH4_VCI_HASH_TYPE == MPIDI_CH4_VCI_HASH_COMM_ONLY) {
```

Comment:
I was curious about the case where a comm gets multiple vci's (coming in a future patch I assume). In that case, how do we decide what vci's get associated with a communicator? Does it get a list of all available vci's or will there by any kind of filtering?

Reply:
The actual hash function determines that.  I don't really like the wording here, but "Hash type" in @rzambre's terminology means that the hash function will only use the `comm` and will ignore the `tag` and `rank`.

Reply:
Yes, for the `COMM_ONLY` selection type, we will have only one VCI per `comm` and the `rank` and `tag` of an operation will be ignored. A `comm` can have multiple VCIs when the selection type is, for example, `COMM_RANK`. The multiple VCIs will get allocated to the `comm` during comm creation. So, the `rank` of an operation would hash into the VCIs stored in the `comm`. I think the number of VCIs to store for a `comm` could be propagated through hints.

Reply:
The case of single vni association (`COMM_ONLY`) was clear to me. I was interested in the latter case (when `comm`, `rank`, and `tag` are used). I see, yea hints can be used to propagate the number of VCIs. Maybe communicator info hints for communicator specific value and a `CVAR` for a global default value.